### PR TITLE
Hero tests 129

### PR DESCRIPTION
**Pull request by steichert**

### CODE REVIEW

File: frontend/app/bot-config/heroes/heroes.component.spec.ts [hunk 1]
```diff
@@ -172,4 +162,56 @@ describe('HeroesComponent', () => {
...
+        fixture.detectChanges();
+        // Change pools to multiple pools
+        fixture.debugElement.query(By.css('#togglePoolsBtn'))
+            .triggerEventHandler('click', new MouseEvent('click'));
+        fixture.detectChanges();
+        expect(component.pools[0][0]['id']).toEqual(component.selectedHeroesList[0]['id']);
+        done();
+    });
+
+    it('should not add duplicates heroes when toggling to single pool', (done) => {
```
[the remark this answers is not in the document]
Comment (steichert) [reply frontend/app/bot-config/heroes/heroes.component.spec.ts]: My bad, I forgot to add that 😅

[the remark this answers is not in the document]
Comment (SpellBreakerZA) [reply frontend/app/bot-config/heroes/heroes.component.spec.ts]: Do you we have a test for adding a duplicate hero to a partitioned pool (like add antimage to pos 2 twice)?

[the remark this answers is not in the document]
Comment (steichert) [reply frontend/app/bot-config/heroes/heroes.component.spec.ts]: No but we can add that as well

[the remark this answers is not in the document]
Comment (SpellBreakerZA) [reply frontend/app/bot-config/heroes/heroes.component.spec.ts]: Also, maybe you should expect the other arrays to be empty to be safe?

[the remark this answers is not in the document]
Comment (SpellBreakerZA) [reply frontend/app/bot-config/heroes/heroes.component.spec.ts]: Ok. Great stuff

**Comment by SpellBreakerZA** [comment]
Do you think it's possible to test the searching feature?

**Comment by steichert** [comment]
> Do you think it's possible to test the searching feature?

I think so, we'd just have to add another hero stub, then search for one and make sure that the other isn't showing?

**Comment by SpellBreakerZA** [comment]
> > Do you think it's possible to test the searching feature?
> 
> I think so, we'd just have to add another hero stub, then search for one and make sure that the other isn't showing?

Could you please do that then?

**Comment by steichert** [comment]
> > > Do you think it's possible to test the searching feature?
> > 
> > 
> > I think so, we'd just have to add another hero stub, then search for one and make sure that the other isn't showing?
> 
> Could you please do that then?

Or would it be enough to just add one hero, and then search for a different name to check that the result is empty?

**Comment by steichert** [comment]
So no testing the searching then?

**Comment by vector67** [comment]
Not for now.

**Pull request merged**